include/rcpputils/asserts.hpp

```diff
@@ -57,25 +57,27 @@ class RCPPUTILS_PUBLIC IllegalStateException : public std::exception
  * Checks that an argument condition passes.
  *
  * \param condition
+ * \param msg
  * \throw std::invalid_argument if the condition is not met.
  */
-inline void require_true(bool condition)
+inline void require_true(bool condition, const std::string & msg = "Invalid argument passed!")
 {
   if (!condition) {
-    throw std::invalid_argument{"Invalid argument passed!"};
+    throw std::invalid_argument{msg.c_str()};
```

Comment:
This can just take the `std::string` directly, that is don't do `.cstr()`.

Reply:
Please check the other cases as well.

Reply:
Only `std::invalid_argument` accepts an `std::string`.